platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcCredentialsResolver.java

```diff
@@ -103,15 +118,16 @@ public void validate(OidcCredentials credentials, WebContext webContext) {
       }
     }
 
-    // try to get credentials using access token
-    final AccessToken accessToken = credentials.getAccessToken();
-    if (credentials.getIdToken() == null && accessToken != null) {
+    // try to get id token using access token
+    if (credentials.getIdToken() == null && initialAccessToken != null) {
+
+      oidcTokenValidator.validateAccessToken(initialAccessToken, null);
```

Comment:
If we do the validation at the top we won't need this call here

Reply:
Right now we are validating all the tokens when we are about to use them, so I think keeping it here for now makes the most sense.

I used your code recommendation about validating the access token at the very beginning, but I added it into the "is the ID token null" condition, so it shouldn't be validated twice:
```
    if (initialIdToken != null) {
      oidcTokenValidator.validateAccessToken(initialAccessToken, initialIdToken);
      oidcTokenValidator.validateIdTokens(initialIdToken, webContext);
      return;
    }
```

Reply:
The thing with that is if we don't use the access token we won't validate it right? I think we should either way